### ISSUE TITLE
"Show potential fixes" is triggered by refactorings unrelated to a diagnostic

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#47636]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1168988/potential-fix-on-cs0029-cannot-implicitly-convert.html)._

---
[severity:I'm somewhat annoyed. A fix would be nice]
IntelliSense correction on CS0029 Cannot implicitly convert type 'string' to 'bool' do not makes sense. It just changes a null object to negate?

---
#### Feedback Bot on 9/1/2020, 01:25 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Vatsalya Agrawal [MSFT] on 9/2/2020, 04:45 PM: 

<p>Thank you for providing feedback, and we’re sorry to hear it’s not behaving as you expect. Based on your description so far, we were unable to reproduce the issue. Could you please share sample code where this issue is occuring.</p>

#### Dinesh Rajan on 9/2/2020, 05:14 PM: 

Attaching a screenshot and recording to see how this works in my system. Please see if this helps to check
#### Feedback Bot on 9/11/2020, 09:54 AM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by sharwell** [comment]
This issue is occurring because the **Show potential fixes** feature of Quick Info is getting triggered by _refactorings_, even when there is no code fix available for the reported diagnostic.

### Steps to reproduce

1. Add the following code to a method:

    ```csharp
    string test = null;
    if (!test)
    {
    }
    ```

2. Place the caret in the middle of `!test`

3. Hover the mouse over `!test`

### Expected results

![image](https://user-images.githubusercontent.com/1408396/92953605-da785b00-f416-11ea-8fb6-af1aa9f4dad8.png)

### Actual results

![image](https://user-images.githubusercontent.com/1408396/92953637-ecf29480-f416-11ea-9be1-e98a2186be6a.png)

### Additional details

The refactoring triggered by caret placement (step 2) is the **Invert If** refactoring, which is unrelated to CS0023.

**Comment by mavasani** [comment]
This was due to a design change mentioned here: https://github.com/dotnet/roslyn/issues/44594#issuecomment-637754997. We decided in the IDE team meeting that the light bulb under the caret (which is meant for show potential fixes), should show all applicable refactorings under the span. For example, if you just edit an existing variable to a new name, we want to show the rename refactoring as a potential fix. If we feel this is noisy, we can scale back and limit the refactorings under the caret to just high priority refactorings.

**Comment by CyrusNajmabadi** [comment]
> If we feel this is noisy, we can scale back and limit the refactorings under the caret to just high priority refactorings.

I don't think this would be ok.  People (including myself) depend on these refatorings *including*, and *commonly* while teh code is in error.  

Note: we could potentially have two verbs here for opening the list.  i.e. a verb to show everything (`ctrl-.`) and one to just show fixes for errors.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback.
